portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -95,6 +98,19 @@ private String getComment(HttpServletRequest request) {
     return request.getParameter("comment");
   }
 
+  //Get request number from query string
+  private int getRequestNum(HttpServletRequest request) {
+    String requestString = request.getParameter("request");
+    int requestNum;
+    try {
+      requestNum = Integer.parseInt(requestString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + requestString);
```

Comment:
Print statements can get swallowed or be difficult to track later. I'm not sure how servlets handle error logs, but you might want to use logging from log4J instead of a print.